Testing/CICDTests/Services/ReadmeServiceTests.cs

```diff
@@ -0,0 +1,417 @@
+    }
+
+    [Theory]
+    [InlineData("C:/test-dir", "C:/test-dir")]
+    [InlineData(@"C:\test-dir", "C:/test-dir")]
+    public void Ctor_WithDifferentDirectorySeparators_ConvertsToCrossPlatformPath(
+        string path,
+        string expected)
+    {
+        // Arrange
+        this.mockFindDirService.Setup(m => m.FindDescendentDir(It.IsAny<string?>(), It.IsAny<string>()))
+            .Returns(path);
+
+        // Act
+        _ = CreateSystemUnderTest();
+
+        // Assert
+        this.mockFindDirService.Verify(m =>
+            m.FindDescendentDir(BaseDirPath, GitHubDirName), Times.Once);
+        this.mockPath.Verify(m => m.GetDirectoryName(expected), Times.Once);
+    }
+    #endregion
+
+    #region Method Tests
+    [Fact]
+    public void RunPreProcessing_WhenFileDoesNotExist_ThrowsException()
+    {
+        // Arrange
+        this.mockFile.Setup(m => m.Exists(It.IsAny<string>())).Returns(false);
+        var sut = CreateSystemUnderTest();
+
+        // Act
+        var act = () => sut.RunPreProcessing();
+
+        // Assert
+        act.Should().ThrowExactly<FileNotFoundException>()
+            .WithMessage($"The readme file '{ReadMeFilePath}' does not exist.");
+    }
+
+    [Theory]
+    [InlineData(" ", "")]
+    [InlineData("  ", "")]
+    [InlineData("", " ")]
+    [InlineData("", "  ")]
+    [InlineData(" ", " ")]
+    [InlineData("  ", "  ")]
+    public void RunPreProcessing_WithNonMatchingCommentStartsAndStops_ThrowsException(
+        string preText,
+        string postText)
+    {
+        // Arrange
+        var fileContent = new[]
+        {
+            $"{preText}{ProcessingCommentStart}{postText}",
+            "content to comment",
+        };
+
+        this.mockFile.Setup(m => m.ReadAllLines(It.IsAny<string>()))
+            .Returns(fileContent);
+
+        var sut = CreateSystemUnderTest();
+
+        // Act
+        var act = () => sut.RunPreProcessing();
+
+        // Assert
+        // TODO: need to use different exception
+        act.Should().ThrowExactly<ReadmeProcessingException>()
+            .WithMessage("Readme file processing must have equal comment block starts and stops.");
+    }
+
+    [Theory]
+    [InlineData(" ", "")]
+    [InlineData("  ", "")]
+    [InlineData("", " ")]
+    [InlineData("", "  ")]
+    [InlineData(" ", " ")]
+    [InlineData("  ", "  ")]
+    public void RunPreProcessing_WithNonMatchingUncommentStartsAndStops_ThrowsException(
+        string preText,
+        string postText)
+    {
+        // Arrange
+        var fileContent = new[]
+        {
+            $"{preText}{ProcessingUncommentStart}{postText}",
+            "content to uncomment",
+        };
+
+        this.mockFile.Setup(m => m.ReadAllLines(It.IsAny<string>()))
+            .Returns(fileContent);
+
+        var sut = CreateSystemUnderTest();
+
+        // Act
+        var act = () => sut.RunPreProcessing();
+
+        // Assert
+        // TODO: need to use different exception
```

Comment:
Is this TODO still needed? 
```suggestion
        // TODO: need to use different exception
```